### ISSUE TITLE
Contact Form: dropdown error message is not accessible

### ISSUE THREAD

**Issue by monsieur-z** [Automattic/jetpack#34253]
### Impacted plugin

Jetpack

### Quick summary

The error message of the dropdown field in the Form package is not accessible. Screen readers do not read it, nor is the input announced as invalid.

### Steps to reproduce

1. Create a new post.
2. Add a form from the Form block.
3. Add a dropdown and mark it as required.
4. Visit the post preview and open a screen reader.
5. Submit the post. You should get an error message.
6. Tabbing back to the dropdown field, notice the screen reader is not reading any error message.

### A clear and concise description of what you expected to happen.

The screen reader should announce that the input is invalid and say the error message.

### What actually happened

The screen reader doesn't announce that the input is invalid and doesn't say the error message.

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### Logs or notes

Video comparing how VoiceOver reads a text input (desired behavior) and the dropdown (faulty behavior).

https://github.com/Automattic/jetpack/assets/1620183/bb12c541-7883-4369-9e71-a6aa2634bcf8

**Comment by monsieur-z** [comment]
It's worth considering dropping jQuery UI select menu in favor of a native `select`, which should be accessible by default.
